tcti/tcti_socket.c

```diff
@@ -528,3 +534,65 @@ TSS2_RC InitSocketTcti (
 
     return rval;
 }
+
+TSS2_RC Tss2_Tcti_Socket_Init (
+    TSS2_TCTI_CONTEXT *tctiContext,
+    size_t *size,
+    const char *conf
+    )
+{
+    TCTI_SOCKET_CONF sock_conf = { 0 };
+    TSS2_RC rc;
+    UriParserStateA state;
+    UriUriA uri;
+    const char *uri_str = conf != NULL ? conf : TCTI_SOCKET_DEFAULT;
+    size_t range;
+    /* maximum 5 digits in uint16_t + 1 for \0 */
+    char port[6] = { 0 };
+    char hostname[HOST_NAME_MAX + 1] = { 0 };
+
+    state.uri = &uri;
+    if (uriParseUriA (&state, uri_str) != URI_SUCCESS) {
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+
+    /* extract host & domain name / fqdn */
+    range = uri.hostText.afterLast - uri.hostText.first;
+    if (range > HOST_NAME_MAX) {
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+    strncpy (hostname, uri.hostText.first, range);
+    sock_conf.hostname = hostname;
+
+    /* extract port number */
+    range = uri.portText.afterLast - uri.portText.first;
+    if (range < 5 && range > 0) {
+        strncpy (port, uri.portText.first, range);
+        sscanf (port, "%" SCNu16, &sock_conf.port);
```

Comment:
if (sscanf(port ...) != EOF)? or will the uriParseUriA() do it?

Reply:
`uriParseUriA` catches some invalid port stuff. It'll return an error if the port string has non-numeric characters in it. But it won't detect port numbers outside the valid range (1 to UINT16_MAX). So if there's anything in the port string (aka `range <= 5 && range > 0`) then it's going to be an unsigned integer and `sscanf` won't return EOF.

All that said checking for the return value makes sense just on the principle of CYA. I'll map these errors to to TSS2_TCTI_RC_BAD_VALUE. PR updated.